docker/ubuntu/Dockerfile

```diff
@@ -45,7 +46,9 @@ ARG GCC_VERSION=12
 RUN apt install -y gcc-${GCC_VERSION} g++-${GCC_VERSION}
 ENV CC=/usr/bin/gcc-${GCC_VERSION}
 ENV CXX=/usr/bin/gcc-${GCC_VERSION}
-RUN rm -rf /var/lib/apt/lists/*
+
+# Clean up unnecessary files to reduce image size.
+RUN rm -rf /var/lib/apt/lists/* && apt autoremove -y && apt clean
```

Comment:
I've done some comparisons and 
```dockerfile
RUN rm -rf /var/lib/apt/lists/*
``` 
is usually enough and `apt autoremove` probably is a noop here.
I guess
```dockerfile
RUN rm -rf /var/lib/apt/lists/* && apt clean
```
if you just want to be _super_ thorough.